econml/utilities.py

```diff
@@ -697,3 +750,96 @@ def predict(self, Xt):
         t = Xt[:, -self.n_T:]
         predictions = [self.model_list[np.nonzero(t[i])[0][0]].predict(X[[i]]) for i in range(len(X))]
         return np.concatenate(predictions)
+
+
+class StatsModelsWrapper:
+    """
+    Helper class to wrap a StatsModels OLS model to conform to the sklearn API.
+
+    Parameters
+    ----------
+    fit_intercept: bool (default False)
+        Whether to fit an intercept
+
+    Attributes
+    ----------
+    fit_args: dict of str: object
+        The arguments to pass to the `OLS` regression's `fit` method.  See the
+        statsmodels documentation for more information.
+    results: RegressionResults
+        After `fit` has been called, this attribute will store the regression results.
+    """
+
+    def __init__(self, fit_intercept=False):
+        self.fit_args = {}
+        self.fit_intercept = fit_intercept
+
+    def fit(self, X, y, sample_weight=None):
+        """
+        Fit the ordinary least squares model.
+
+        Parameters
+        ----------
+        X : array-like, shape (n_samples, n_features)
+            Training data
+        y : array_like, shape (n_samples, 1) or (n_samples,)
+            Target values
+        sample_weight : array_like, shape (n_samples,)
+            Individual weights for each sample
+
+        Returns
+        -------
+        self
+        """
+        assert ndim(y) == 1 or (ndim(y) == 2 and shape(y)[1] == 1)
+        y = reshape(y, (-1,))
+        if self.fit_intercept:
+            X = add_constant(X, has_constant='add')
+        if sample_weight is not None:
```

Comment:
It is better to use WLS instead of OLS. In fact OLS doesn't seem to have documented that it even accepts sample_weights and that it treats them correctly. See here:
https://www.statsmodels.org/devel/generated/statsmodels.regression.linear_model.WLS.html#statsmodels.regression.linear_model.WLS
https://www.statsmodels.org/devel/generated/statsmodels.regression.linear_model.OLS.html

Also better to state to the statsmodel that we have added a constant. So maybe we should change to the following:
```suggestion
        if sample_weight is not None:
            ols = WLS(y, X, weights=sample_weight, hasconst=self.fit_intercept)
        else:
            ols = WLS(y, X, hasconst=self.fit_intercept)
```

Reply:
`OLS` inherits from `WLS` (for some reason...), but I agree that it would be better not to rely on this so I'll switch to `WLS`.